src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java

```diff
@@ -179,7 +179,8 @@ public class AllChecksTest extends AbstractModuleTestSupport {
                 "ARRAY_INIT",
                 "ELLIPSIS",
                 // these are covered by GenericWhitespaceCheck
-                "WILDCARD_TYPE", "GENERIC_END", "GENERIC_START")
+                "WILDCARD_TYPE", "GENERIC_END", "GENERIC_START",
+                "LITERAL_WHEN")
```

Comment:
Show me that when is covered here.

Reply:
can you restate. I don't understand. what do you mean by "covered here"

Reply:
@mahfouz72 what would happen if we don't exclude this token? It should not appear in our code until we migrate to java 21.

Reply:
there is a test that validate that all checks tokens are used in google config https://github.com/checkstyle/checkstyle/blob/42158b1fbad0ee3ef2d60a246635c805910c4057/src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java#L391

Reply:
So we should add it to the config or exclude it like I did to make the test passes

Reply:
added `LITERAL_WHEN` to the config.

Reply:
@mahfouz72 What is your understanding of what it means to put the token here at this location?

Reply:
Ideally, we should have all checks tokens in our checkstyle config and we have a test to enforce that. This list is to exclude some tokens from being added in the config (suppress the test for this specific token). an example for this `GENERIC_START` which should be in the config of the whitespaceAround check but we didn't add it there because we have a separate check for it so we added it in this exclude list

Reply:
My understanding at first is why we would need to add LITERAL_WHEN in the config? we don't use this syntax in our code what will we gain if we add it in the config? that's why I excluded it

Reply:
Please correct me if anything is wrong and clarify this comment https://github.com/checkstyle/checkstyle/pull/15311#discussion_r1685632454

Reply:
So this area is to document what tokens are excluded from our config as well as the other configs. When we add these tokens here, we document why. The comment that is next to the token you added is `these are covered by GenericWhitespaceCheck`. To me, that comment coveys why we are adding this token here. So my question at the time was, how does `GenericWhitespaceCheck` cover this token for another check to satisfy why we aren't adding it.

Reply:
ahh, The comment was misleading I didn't want to say that it applies to `LITERAL_WHEN`. actually there was another comment (https://github.com/checkstyle/checkstyle/pull/15311#discussion_r1683363896)    `// 'when' keyword can be used by newer java versions only` but I removed it based on @nrmancuso review he was probably was trying to say to remove the token at all from the list but i removed the comment only

Reply:
Okay now we are good I added the token to the config and removed it from the exclude list. I just was confused why we added to our config a token that we are not using. but as @nrmancuso said [here](https://github.com/checkstyle/checkstyle/pull/15311#discussion_r1685742936) there should be no issues with this. we will use it when we migrate to java 17